include/gna.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * Author: Sathish Kuttan <sathish.k.kuttan@intel.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API header file for Intel GNA driver
+ *
+ * This file contains the driver APIs for Intel's
+ * Gaussian Mixture Model and Neural Network Accelerator (GNA)
+ */
+
+#ifndef __INCLUDE_GNA__
+#define __INCLUDE_GNA__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * GNA driver configuration structure.
+ * Currently empty.
+ */
+struct gna_config {
+};
+
+/**
+ * GNA Neural Network model header
+ * Describes the key parameters of the neural network model
+ */
+struct gna_model_header {
+	u32_t	labase_offset;
+	u32_t	model_size;
+	u32_t	gna_mode;
+	u32_t	layer_count;
+	u32_t	bytes_per_input;
+	u32_t	bytes_per_output;
+	u32_t	num_input_nodes;
+	u32_t	num_output_nodes;
+	u32_t	input_ptr_offset;
+	u32_t	output_ptr_offset;
+	u32_t	rw_region_size;
+	u32_t	input_scaling_factor;
+	u32_t	output_scaling_factor;
+};
+
+/**
+ * GNA Neural Network model information to be provided by application
+ * during model registration
+ */
+struct gna_model_info {
+	struct gna_model_header *header;
+	void *rw_region;
+	void *ro_region;
+};
+
+/**
+ * Request to perform inference on the given neural network model
+ */
+struct gna_inference_req {
+	void *model_handle;
+	void *input;
+	void *output;
+	void *intermediate;
+};
+
+/**
+ * Statistics of the inference operation returned after completion
+ */
+struct gna_inference_stats {
+	u32_t total_cycles;
+	u32_t stall_cycles;
+	u32_t cycles_per_sec;
+};
+
+/**
+ * Result of an inference operation
+ */
+enum gna_result {
+	GNA_RESULT_INFERENCE_COMPLETE,
+	GNA_RESULT_SATURATION_OCCURRED,
+	GNA_RESULT_OUTPUT_BUFFER_FULL_ERROR,
+	GNA_RESULT_PARAM_OUT_OF_RANGE_ERROR,
+	GNA_RESULT_GENERIC_ERROR,
+};
+
+/**
+ * Structure containing a response to the inference request
+ */
+struct gna_inference_resp {
+	enum gna_result result;
+	void *output;
+	size_t output_len;
+	struct gna_inference_stats stats;
+};
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * Internal documentation. Skip in public documentation
+ */
+typedef int (*gna_callback)(struct gna_inference_resp *result);
+
+typedef int (*gna_api_config)(struct device *dev, struct gna_config *cfg);
+typedef int (*gna_api_register)(struct device *dev,
+		struct gna_model_info *model, void **model_handle);
+typedef int (*gna_api_deregister)(struct device *dev, void *model_handle);
+typedef int (*gna_api_infer)(struct device *dev, struct gna_inference_req *req,
+		gna_callback callback);
+
+struct gna_driver_api {
+	gna_api_config		configure;
+	gna_api_register	register_model;
+	gna_api_deregister	deregister_model;
+	gna_api_infer		infer;
+};
+
+/**
+ * @endcond
+ */
+
```

Comment:
You will need to document the API with doxygen decorations.

Reply:
Good catch. Done.